extensions/amp-form/0.1/amp-form.js

```diff
@@ -88,11 +92,14 @@ export class AmpForm {
         method: this.method_,
         credentials: 'include',
         requireAmpResponseSourceOrigin: true,
-      }).then(() => this.setState_(FormState_.SUBMIT_SUCCESS))
-          .catch(error => {
-            this.setState_(FormState_.SUBMIT_ERROR);
-            rethrowAsync('Form submission failed:', error);
-          });
+      }).then(response => {
+        this.setState_(FormState_.SUBMIT_SUCCESS);
+        this.renderTemplate_(FormState_.SUBMIT_SUCCESS, response);
+      }).catch(error => {
+        this.setState_(FormState_.SUBMIT_ERROR);
+        this.renderTemplate_(FormState_.SUBMIT_ERROR, error);
```

Comment:
Yeah, we don't currently pass real data and just use the error instance that xhr-impl throws. The current xhr setup ignores the body of errors statuses. 

Can we change the implementation for fetchJson? Instead of throwing an error in `xhr.assertSuccess` we instead return a rejected promise with the response?

Would this be a way to do this?
```
    return this.fetchAmpCors_(input, init).then(response => {
      if (!isSuccess(response)) {
        return Promise.reject({error: new Error('message'), response: response});
      }
      return response.json();
    })
```

And then the invoker would do something like the following:
```javascript
      fetchJson().then(...).catch(errorData => {
        this.setState_(FormState_.SUBMIT_ERROR);
        errorData.response.json().then(data =>
            this.renderTemplate_(FormState_.SUBMIT_ERROR, data));
        rethrowAsync('Form submission failed:', errorData.error);
      });
```

What do you think?